shared/js/manifest_helper.js

```diff
@@ -51,6 +51,10 @@ function ManifestHelper(manifest) {
   }
 }
 
+ManifestHelper.prototype.getShortNameOrName = function() {
```

Comment:
I think a displayName getter would work. Both from a security and UX point of view, the concern was that we should be consistent in how we represent an app - at install time, in use, in settings etc. When we went looking for where in Gaia we would want .name would be surfaced instead of .short_name when it was defined, the answer was "nowhere".  manifest.displayName then seems to describe what we want while leaving the option to get .name if that's what you really mean.